src/btree/bt_walk.c

```diff
@@ -528,7 +528,8 @@ __tree_walk_internal(WT_SESSION_IMPL *session, WT_REF **refp, uint64_t *walkcntp
 done:
 err:
     WT_TRET(__wt_page_release(session, couple, flags));
-    WT_TRET(__wt_page_release(session, ref_orig, flags));
+    if (!LF_ISSET(WT_READ_NO_PAGE_RELEASE))
+        WT_TRET(__wt_page_release(session, ref_orig, flags));
     WT_LEAVE_PAGE_INDEX(session);
```

Comment:
I have a feeling that the WT_LEAVE_PAGE_INDEX needs to come after the `wt_page_release` call. Otherwise something else can reclaim the PINDEX structure this thread is relying on, which can lead to memory corruption.

Reply:
(it's a feeling because I didn't look at the LEAVE_PAGE_INDEX macro to double check that's what it does.